codegenerator/cli/templates/static/codegen/src/eventFetching/FetchState.res

```diff
@@ -1025,9 +1025,14 @@ let rec rollbackRegisterList = (
 
 let rollback = (self: t, ~lastKnownValidBlock) => {
   let baseRegister = rollbackRegisterList(self.baseRegister, ~lastKnownValidBlock)
-  //TODO rollback pending dynamic contract registrations
+
+  let pendingDynamicContracts =
+    self.pendingDynamicContracts->Array.keep(({registeringEventBlockNumber}) =>
+      registeringEventBlockNumber <= lastKnownValidBlock.blockNumber
```

Comment:
Should it also consider logIndex?

Reply:
No since when we rollback we always do a full block. If you look at the filters we add: 

```res
        cf
        ->ChainFetcher.rollbackToLastBlockHashes(~rolledBackLastBlockData)
        ->ChainFetcher.addProcessingFilter(
          ~filter=eventBatchQueueItem => {
            let {timestamp, blockNumber} = eventBatchQueueItem
            //Filter out events that occur passed the block where the query starts but
            //are lower than the timestamp where we rolled back to
            (timestamp, chain->ChainMap.Chain.toChainId, blockNumber) >
            (lastKnownValidBlockTimestamp, rollbackChainId, lastKnownValidBlockNumber)
          },
          ~isValid=(~fetchState) => {
            //Remove the event filter once the fetchState has fetched passed the
            //timestamp of the valid rollback block's timestamp
            let {blockTimestamp, blockNumber} = FetchState.getLatestFullyFetchedBlock(fetchState)
            (blockTimestamp, chain->ChainMap.Chain.toChainId, blockNumber) <=
            (lastKnownValidBlockTimestamp, rollbackChainId, lastKnownValidBlockNumber)
          },
        )
```

We only care about the block number